projects/plugins/social/tests/e2e/helpers/connection.ts

```diff
@@ -0,0 +1,29 @@
+import { expect, Page } from '@playwright/test';
+import logger from '_jetpack-e2e-commons/logger.js';
+
+/**
+ * Connect Jetpack Social
+ * @param {Page}    page    - Playwright page object
+ * @param {boolean} premium - If true, selects the Social premium plan and proceeds with purchase steps; if false, selects the free plan.
+ */
+export async function connect( page: Page, premium: boolean = false ) {
+	logger.debug( 'Connect Jetpack Social' );
+
+	await page.goto( '/wp-admin/admin.php?page=jetpack-social' );
+	await page.getByRole( 'button', { name: 'Get Started' } ).click();
+
+	await expect( page.locator( 'button > svg.components-spinner' ) ).toBeHidden( {
+		timeout: 40000,
+	} );
```

Comment:
this approach is not very intuitive and has an unusual timeout which results in flakey tests. May be we can use this approach 

https://github.com/Automattic/jetpack/blob/291910b36c7fe1b397653b904612d7ede4c4a8ea/projects/plugins/jetpack/tests/e2e/helpers/onboarding.ts#L99-L104